Update to lotus2 2.07

hildebra opened the pull request:
Also:
- Add -sdmopt optional input
- Make `-chim_skew`, `-id`, `-LCA_cover` and `-LCA_frac` optional
- Add "(Default)" option to `-ref_db`, `-deactivateChimeraCheck`, `-platform`, `-amplicon_type` and `-tax_group`

Commit:
t1

nsoranzo commented on tools/lotus2/lotus2.xml:
```diff
@@ -10,7 +10,8 @@
                 <when value="cached">
                     <param argument="ref_db" type="select" label="Using reference database" help="Select database from the list">
-                        <option value="SLV" selected="true">Silva LSU (23/28S) or SSU (16/18S) (SLV)</option>
+                        <option value="" style="color:gray" selected="true">(Default)</option>
```
I don't think `style` is a thing in Galaxy tool XML (same below).
```suggestion
                        <option value="" selected="true">(Default)</option>
```

nsoranzo commented on tools/lotus2/lotus2.xml:
```diff
@@ -258,7 +284,8 @@ exit \$EXIT_VALUE
                     <option value="LSU">LSU Large subunit (23S/28S)</option>
-                    <option value="SSU" selected="true">SSU small subunit (16S/18S)</option>
+                    <option value="SSU">SSU small subunit (16S/18S)</option>
+                    <option value="" style="color:gray" selected="true">(Default)</option>
```
I'd put this option first.

hildebra replied:
ok, but isn't it last currently?

nsoranzo replied:
SSU (the current default) is in the middle because I think "LSU, SSU, ITS, ITS1, ITS2" was a good order, but the new "(Default)" should go first. I've added suggestions to implement that.
```suggestion
```

nsoranzo commented on tools/lotus2/lotus2.xml:
```diff
@@ -106,24 +109,30 @@ lotus2
     -utax_thr $tax_args.aligner_cond.utax_thr
 #else:
-    #if $tax_args.aligner_cond.refDB_cond.refDB == 'cached':
+    #if $tax_args.aligner_cond.refDB_cond.refDB == '':
+    #elif $tax_args.aligner_cond.refDB_cond.refDB == 'cached':
         -refDB $tax_args.aligner_cond.refDB_cond.ref_db
         -greengenesSpecies $tax_args.aligner_cond.refDB_cond.greengenesSpecies
```
```suggestion
    #if $tax_args.aligner_cond.refDB_cond.refDB == 'cached':
        #if $tax_args.aligner_cond.refDB_cond.ref_db != '':
            -refDB $tax_args.aligner_cond.refDB_cond.ref_db
            -greengenesSpecies $tax_args.aligner_cond.refDB_cond.greengenesSpecies
        #end if
```

nsoranzo commented on tools/lotus2/lotus2.xml:
```diff
@@ -171,7 +195,8 @@ exit \$EXIT_VALUE
         <param argument="-map" type="data" format="tabular" optional="true" label="Mapping file (optional)" help="Needed to demultiplex the FASTQ files using sdm. If the FASTQ are already demultiplexed, this can be omitted." />
         <param argument="-platform" type="select" label="Sequencing platform">
-            <option value="miSeq" selected="true">miSeq</option>
+            <option value="" style="color:gray" selected="true">(Default)</option>
```
```suggestion
            <option value="" selected="true">(Default)</option>
```

hildebra replied:
There's a difference between not selecting anything, and selecting miSeq (the difference being that in the first case lotus attempts to figure it out on its own)

nsoranzo replied:
I understand, I am just suggesting to remove the `style` here?

nsoranzo commented on tools/lotus2/lotus2.xml:
```diff
@@ -106,24 +109,30 @@ lotus2
 -clustering $clu_args.clu_cond.clustering
-#if $clu_args.clu_cond.clustering in ('1', '3'):
-    -id $clu_args.clu_cond.id
+#if ($clu_args.clu_cond.clustering in ('1', '3') ) and ( $clu_args.clu_cond.id != 0.97 ):
```
What difference does this change make?

hildebra replied:
doesn't give the value to lotus2. You have to avoid giving command args to lotus where possible.. there is an intricate system of choosing best default values, based on other parameters, and to take advantage of this system (and not to replicate it again in galaxy), you need to not give flags to the lotus2 call

nsoranzo replied:
Please don't do that, it's an unexpected behaviour for the user to see a value that is then not passed to the program.
For integer/float parameters, the right solution is to make the input optional and leave the value empty, see suggestions. Note that the easiest way to check in Cheetah if an optional integer/float parameter has been set by the user is to cast it to str.
```suggestion
#if $clu_args.clu_cond.clustering in ('1', '3') and str($clu_args.clu_cond.id):
```

nsoranzo commented on tools/lotus2/lotus2.xml:
```diff
@@ -106,24 +109,30 @@ lotus2
     -swarm_distance $clu_args.clu_cond.swarm_distance
 #end if
-#if $clu_args.derepMin:
+#if ( $clu_args.derepMin ) and ( $clu_args.derepMin != '' ) :
```
That's already included in the original test (the empty string is false).

```suggestion
#if $clu_args.derepMin:
```

nsoranzo commented on tools/lotus2/lotus2.xml:
```diff
@@ -223,7 +248,8 @@ exit \$EXIT_VALUE
             <param argument="-derepMin" type="text" value="" label="Minimum size of dereplicated raw reads (optional)" help="E.g. 4:1,4:2,3:3 . See http://lotus2.earlham.ac.uk/images/Derep_options.pdf for how to specify this parameter. If not specified, LotuS2 will select an appropriate default for the chosen clustering algorithm." />
             <param argument="-deactivateChimeraCheck" type="select" label="Chimera check">
-                <option value="0" selected="true">OTU chimera checks</option>
+                <option value="" style="color:gray" selected="true">(Default)</option>
```
```suggestion
                <option value="" selected="true">(Default)</option>
```

nsoranzo commented on tools/lotus2/lotus2.xml:
```diff
@@ -106,24 +109,30 @@ lotus2
 #end if
+
+#if $clu_args.deactivateChimeraCheck != '':
 -deactivateChimeraCheck $clu_args.deactivateChimeraCheck
```
```suggestion
    -deactivateChimeraCheck $clu_args.deactivateChimeraCheck
```

nsoranzo commented on tools/lotus2/lotus2.xml:
```diff
@@ -274,7 +301,8 @@ exit \$EXIT_VALUE
             </conditional>
             <param argument="-tax_group" type="select" label="Tax group">
-                <option value="bacteria" selected="true">bacterial 16S rDNA annnotation</option>
+                <option value="" style="color:gray" selected="true">(Default)</option>
```
```suggestion
                <option value="" selected="true">(Default)</option>
```

Commit:
Update tools/lotus2/lotus2.xml

Co-authored-by: Nicola Soranzo <nicola.soranzo@gmail.com>

Commit:
Update tools/lotus2/lotus2.xml

Co-authored-by: Nicola Soranzo <nicola.soranzo@gmail.com>

Commit:
Update tools/lotus2/lotus2.xml

Co-authored-by: Nicola Soranzo <nicola.soranzo@gmail.com>

Commit:
Update tools/lotus2/lotus2.xml

Co-authored-by: Nicola Soranzo <nicola.soranzo@gmail.com>

Commit:
Update tools/lotus2/lotus2.xml

Co-authored-by: Nicola Soranzo <nicola.soranzo@gmail.com>

Commit:
Update tools/lotus2/lotus2.xml

Co-authored-by: Nicola Soranzo <nicola.soranzo@gmail.com>

nsoranzo commented on tools/lotus2/lotus2.xml:
```diff
@@ -258,7 +285,8 @@ exit \$EXIT_VALUE
             <conditional name="amplicon_cond">
                 <param argument="-amplicon_type" type="select" label="Amplicon type">
                     <option value="LSU">LSU Large subunit (23S/28S)</option>
```
```suggestion
                    <option value="" selected="true">(Default)</option>
                    <option value="LSU">LSU Large subunit (23S/28S)</option>
```

Commit:
Update tools/lotus2/lotus2.xml

Co-authored-by: Nicola Soranzo <nicola.soranzo@gmail.com>

nsoranzo commented on tools/lotus2/lotus2.xml:
```diff
@@ -27,7 +28,7 @@
         </xml>
         <xml name="id_macro">
-            <param argument="-id" type="float" min="0" max="1" value="0.97" label="Clustering threshold for OTUs" />
+            <param argument="-id" type="float" min="0.9" max="1" value="0.97" label="Clustering threshold for OTUs" />
```
```suggestion
            <param argument="-id" type="float" min="0.9" max="1" value="" optional="true" label="Clustering threshold for OTUs" />
```

nsoranzo commented on tools/lotus2/lotus2.xml:
```diff
@@ -106,16 +109,21 @@ lotus2
-#if $clu_args.clu_cond.clustering in ('1', '3'):
-    -id $clu_args.clu_cond.id
+#if ($clu_args.clu_cond.clustering in ('1', '3') ) and ( $clu_args.clu_cond.id != 0.97 ):
+    -id $clu_args.clu_cond.id 
```
```suggestion
    -id $clu_args.clu_cond.id
```

Commit:
Update tools/lotus2/lotus2.xml

Co-authored-by: Nicola Soranzo <nicola.soranzo@gmail.com>

Commit:
Update tools/lotus2/lotus2.xml

Co-authored-by: Nicola Soranzo <nicola.soranzo@gmail.com>

Commit:
Update tools/lotus2/lotus2.xml

Co-authored-by: Nicola Soranzo <nicola.soranzo@gmail.com>

Commit:
Update tools/lotus2/lotus2.xml

Co-authored-by: Nicola Soranzo <nicola.soranzo@gmail.com>

Commit:
Update tools/lotus2/lotus2.xml

Co-authored-by: Nicola Soranzo <nicola.soranzo@gmail.com>

nsoranzo committed:
Update to lotus2 2.07

nsoranzo committed:
Make `-chim_skew`, `-LCA_cover` and `-LCA_frac` optional

Restore `-lulu` and `-buildPhylo`

nsoranzo committed:
Add `-sdmopt` optional input

nsoranzo committed:
Fix typo

nsoranzo committed:
Make clear that `-sdmopt` is optional

nsoranzo committed:
Pass custom `-refDB` FASTA file as a symlink

Otherwise the aligner tries to create index files alongside the .dat
file. Also LAMBDA requires the correct file extension.

nsoranzo committed:
Allow fasta.gz as input format when possible
